### PR TITLE
[spirv] stop compilation if decl evaluation results in an error.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp [hunk 1]
```diff
@@ -580,6 +580,9 @@ void SpirvEmitter::HandleTranslationUnit(ASTContext &context) {
 
   // The entry function is the seed of the queue.
   for (auto *decl : tu->decls()) {
+    if (context.getDiagnostics().hasErrorOccurred())
```
Comment (jaebaek) [inline tools/clang/lib/SPIRV/SpirvEmitter.cpp]: nit: why don't we put it at the end of for-loop?

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp [hunk 2]
```diff
@@ -605,6 +608,8 @@ void SpirvEmitter::HandleTranslationUnit(ASTContext &context) {
   // The queue can grow in the meanwhile; so need to keep evaluating
   // workQueue.size().
   for (uint32_t i = 0; i < workQueue.size(); ++i) {
+    if (context.getDiagnostics().hasErrorOccurred())
```
Comment (jaebaek) [inline tools/clang/lib/SPIRV/SpirvEmitter.cpp]: Likewise. (and remove line 618?)

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp (hunk 1 above)
Comment (ehsannas) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp]: done

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp (hunk 2 above)
Comment (ehsannas) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp]: done

**Pull request merged**